### PR TITLE
fix VS code filtering

### PR DESCRIPTION
**Pull request by metalogical**
fixes kiteco/kiteco#9169

Based on my investigation, I've come to the conclusion that when a replacement range is specified, the `filterText` is compared with the text under the replacement range. A couple people in the depths of internet discussion seem to also have come to this conclusion.

Unfortunately [we don't have complete control over the sort order](https://github.com/kiteco/kiteco/issues/9049#issuecomment-548985916), so now that we see dict completions, this happens:

Screen Shot 2019-11-01 at 4 50 56 PM
Screen Shot 2019-11-01 at 4 51 11 PM

As far as I can surmise, this is because VS Code's completion fuzzy matching algorithm considers the dict completion a "better match," so they end up at the top. This is particularly egregious in the latter case, since that key doesn't actually exist yet.

So there is a product question (@plungg @its-dhung) around whether we want to 
1. not show completions like `.f` -> `["foo"]
2. show them and accept the poor ordering
3. try to not show them where the dict key doesn't already exist, and accept the poor ordering otherwise
    * this is tricky, so I'd vote to go with option (1) or (2) for now, and create a separate issue for this if need be

Even if we want to do (1), I would argue that we should still merge this fix (since this is more correct), and handle filtering out the corresponding dict completions in `kited`.

cc @Moe-Kite

### CODE REVIEW

File: src/completion.js [hunk 1]
```diff
@@ -77,13 +77,17 @@ const processCompletion = (
 ) => {
```
Comment (edzkite) [inline src/completion.js]: Let's remove `filterText` from the function calls if its now unused.

**Comment by plungg** [comment]
We want Kite to be perceived as a smart autocomplete, which means not showing completions that are invalid or out of place (too high, priority-wise). I expect more people to find `.f` -> `["foo"]` out of place than those that expected it, so I'd reduce this false positive by going for option 1, even if option 3 was doable.

**Comment by metalogical** [comment]
@plungg what about the dataframe case:

Screen Shot 2019-11-05 at 1 13 57 PM

do we also want to not show that completion?

**Comment by plungg** [comment]
@metalogical I think I'd keep it consistent so would have that off as well. Users have the opportunity to select `["key-1"]` when they type `df`

**Pull request merged**